Coveralls integration broken

Apparently since May 2016 https://coveralls.io/github/HadoopGenomics/Hadoop-BAM?branch=master

![](https://cl.ly/2E331u1d3e1H/Screen%20Shot%202017-04-28%20at%2012.18.31%20PM.png)

[My latest "successful" Travis build shows coveralls error](https://travis-ci.org/HadoopGenomics/Hadoop-BAM/builds/212927762#L8449-L8463):

```
[ERROR] Failed to execute goal org.eluder.coveralls:coveralls-maven-plugin:4.1.0:report (default-cli) on project hadoop-bam: I/O operation failed: No coverage report files found -> [Help 1]
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -13,7 +13,9 @@ before_install:
   - sed -e "s/^\\(127\\.0\\.0\\.1.*\\)/\\1 $(hostname | cut -c1-63)/" /etc/hosts | sudo tee /etc/hosts
   - cat /etc/hosts # optionally check the content *after*
 
+script: mvn clean test jacoco:report
+
 after_success:
   - python scripts/deploy/addServerToM2Settings.py
-  - mvn clean deploy -DskipTests=true --settings ~/.m2/mySettings.xml
-  - mvn clean test jacoco:report coveralls:report
+  - mvn coveralls:report
+  - mvn deploy -DskipTests=true --settings ~/.m2/mySettings.xml

--- a/pom.xml
+++ b/pom.xml
@@ -423,7 +423,7 @@
                     <plugin>
                         <groupId>org.jacoco</groupId>
                         <artifactId>jacoco-maven-plugin</artifactId>
-                        <version>0.7.5.201505241946</version>
+                        <version>0.7.6.201602180812</version>
                         <executions>
                             <execution>
                                 <id>prepare-agent</id>
@@ -436,7 +436,15 @@
                     <plugin>
                         <groupId>org.eluder.coveralls</groupId>
                         <artifactId>coveralls-maven-plugin</artifactId>
-                        <version>4.1.0</version>
+                        <version>4.3.0</version>
+                    </plugin>
+                    <plugin>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <version>2.20</version>
+                        <configuration>
+                            <!-- Needed for jacoco data to be written, cf. http://stackoverflow.com/a/25378458/544236 -->
+                            <argLine>${argLine}</argLine>
+                        </configuration>
                     </plugin>
                 </plugins>
             </build>

